packages/smart-forms-renderer/src/stories/itemTypes/Boolean.stories.tsx

```diff
@@ -37,24 +36,112 @@ export default meta;
 
 // More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
+const targetText = 'Have you eaten yet?';
+const targetlinkId = 'eaten';
+
+const basicQuestionnare = questionnaireFactory([{
+  linkId: targetlinkId,
+  type: 'boolean',
+  repeats: false,
+  text: targetText
+}])
+
+const basicQr = qrFactory([{
+  linkId: targetlinkId,
+  text: targetText,
+  answer: [
+    {
+      valueBoolean: true
+    }
+  ]
+}])
 
 export const BooleanBasic: Story = {
   args: {
-    questionnaire: qBooleanBasic
+    questionnaire: basicQuestionnare
+  }, play: async ({ canvasElement }) => {
+    await chooseRadioOption(canvasElement, targetlinkId);
+
+    const result = await getAnswers(targetlinkId);
+    expect(result).toHaveLength(1);
+    expect(result[0]).toEqual(expect.objectContaining({ valueBoolean: true }));
+
+    // Clear value
+    const clear = canvasElement.querySelector('button#clear');
+    fireEvent.click(clear as HTMLElement);
+
+    const resultAfterDelete = await getAnswers(targetlinkId);
+    expect(resultAfterDelete).toHaveLength(0);
   }
-};
+}
 
 export const BooleanBasicResponse: Story = {
   args: {
-    questionnaire: qBooleanBasic,
-    questionnaireResponse: qrBooleanBasicResponse
+    questionnaire: basicQuestionnare,
+    questionnaireResponse: basicQr
+  }, play: async ({ canvasElement }) => {
+    const inputText = await getInputText(canvasElement, targetlinkId);
+
+    expect(inputText).toBe("true")
+  }
+};
+
+const qBooleanCheckbox = questionnaireFactory([{
+  extension: [
+    {
+      url: 'http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl',
+      valueCodeableConcept: {
+        coding: [
+          {
+            system: 'http://hl7.org/fhir/questionnaire-item-control',
+            code: 'check-box'
+          }
+        ]
+      }
+    }
+  ],
+  linkId: 'eaten',
+  type: 'boolean',
+  repeats: false,
+  text: 'Have you eaten yet?'
+}])
+const qrBooleanCheckboxResponse = qrFactory([{
+  linkId: 'eaten',
+  text: 'Have you eaten yet?',
+  answer: [
+    {
+      valueBoolean: true
+    }
+  ]
+}])
+
+export const BooleanCheckboxBasic: Story = {
+  args: {
+    questionnaire: qBooleanCheckbox,
+  }, play: async ({ canvasElement }) => {
+    await chooseCheckBox(canvasElement, targetlinkId)
+
+    const result = await getAnswers(targetlinkId);
+    expect(result).toHaveLength(1);
+    expect(result[0]).toEqual(expect.objectContaining({ valueBoolean: true }));
+
+    // Clear value
+    const clear = canvasElement.querySelector('button#clear');
+    fireEvent.click(clear as HTMLElement);
+
+    const resultAfterDelete = await getAnswers(targetlinkId);
+    expect(resultAfterDelete).toHaveLength(0);
   }
 };
 
 export const BooleanCheckboxResponse: Story = {
   args: {
     questionnaire: qBooleanCheckbox,
     questionnaireResponse: qrBooleanCheckboxResponse
+  }, play: async ({ canvasElement }) => {
+    const input = await getInput(canvasElement, targetlinkId);
```

Comment:
getInput should return the first input (based on the name), so you can write `expect(input).toBeChecked()`